ipatests/test_integration/test_acme.py

```diff
@@ -252,3 +298,55 @@ def test_disable_acme_service(self):
             ['curl', '--fail', self.acme_server],
             ok_returncode=22,
         )
+        result = self.master.run_command(['ipa-acme-manage', 'status'])
+        assert result.returncode == 0
+        assert 'disabled' in result.stdout_text
+
+    ######################
+    # Require ipa-ca SAN
+    ######################
+
+    @server_install_teardown
+    def test_third_party_certs(self):
+        """Require ipa-ca SAN on replacement web certificates"""
+
+        result = self.master.run_command(['ipa-acme-manage', 'enable'])
+        assert result.returncode == 0
+
+        self.create_pkcs12('ca1/server')
+        self.prepare_cacert('ca1')
+
+        # Re-install the existing Apache certificate that has a SAN to
+        # verify that it will be accepted.
+        pin = self.master.get_file_contents(
+            paths.HTTPD_PASSWD_FILE_FMT.format(host=self.master.hostname)
+        )
+        result = self.certinstall(
+            certfile=paths.HTTPD_CERT_FILE,
+            keyfile=paths.HTTPD_KEY_FILE,
+            pin=pin
+        )
+        assert result.returncode == 0
+
+        # Install using a 3rd party cert with a missing SAN for ipa-ca
+        # which should be rejected.
+        result = self.certinstall()
+        print(result.stdout_text)
+        print(result.stderr_text)
+        assert result.returncode == 1
+
+        result = self.master.run_command(['ipa-acme-manage', 'disable'])
+        assert result.returncode == 0
+
+        # Install using a 3rd party cert with a missing SAN for ipa-ca
+        # which should be ok since ACME is disabled.
+        result = self.certinstall()
+        print(result.stdout_text)
+        print(result.stderr_text)
+        assert result.returncode == 0
+
+        # Enable ACME which should fail since the Apache cert lacks the SAN
+        result = self.master.run_command(['ipa-acme-manage', 'enable'],
+                                         raiseonerr=False)
+        assert result.returncode == 1
```

Comment:
Shall we make `ipa-acme-manage enable` command more robust in this case? cause return code 1 implies `"CA is not installed on this server."`

Reply:
I'll look into a new return value for this and ipa-server-certinstall for invalid certificates.

Reply:
I think I'm going to adjust the return values in ipa-acme-manage 0 is success, 1 is an error, and 2 and 3 replace the current 1/2. ipa-server-certinstall only returns 0 == ok, 1 == error. Since ACME hasn't shipped in an official release IMHO it's fine to change this.

Reply:
I'm fine with that change also.